File: 백준/Silver/13241. 최소공배수/최소공배수.py
import sys

def lcm(n, k):
    temp = n * k
    while k > 0:
        n, k = k, n % k
    return temp // n

a, b = map(int, sys.stdin.readline().split())
if a < b:
    a, b = b, a
print(lcm(a, b))

File: 백준/Silver/13241. 최소공배수/최소공배수.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		long a = sc.nextLong(), b = sc.nextLong();
		System.out.println(lcm(a, b));
		
		sc.close();
	}

	static long lcm(long a, long b) {
		return (a * b) / gcd(a, b);
	}

	static long gcd(long a, long b) {
		while (b > 0) {
			long temp = a;
			a = b;
			b = temp % b;
		}
		return a;
	}
}